### PR TITLE
NETBEANS-1935: Add common shortcus to the empty editor view

### PR DESCRIPTION
**Pull request by Chris2011**
for new users to NetBeans to help onboarding them to their first use

How it looks
image

How it reacts to keybinding and profile changes
netbeans-shortcuts-update

Original Ticket: https://issues.apache.org/jira/browse/NETBEANS-1935

### CODE REVIEW

**Comment by Chris2011** [comment]
I dunno whether this is correct with the friend API package and to read out the config file but this is all what I could find. If you know a better way, please lemme know :)

**Comment by Chris2011** [comment]
Strange, locally everything was fine and I could build and also run the application. Smth seems missing here 🤔

**Comment by lkishalmi** [comment]
Well, the dependencies are a bit wrong here.

First core.windows shall not have a dependency on the editor (that is defined in the ide cluster)

Better approach would be is to create an API interface in core windows like an `EditorPaneInfoProvider` or something which can return a Panel that the core.windows could render on the EditorPane when there is no open document.

Use Lookup to get the instance for that provider in `EditorView`. Then create a separate module probably in the nb cluster to provide an implementation of the `EditorPaneInfoProvider`.

**Comment by Chris2011** [comment]
Thx @lkishalmi I will have a look

**Comment by Chris2011** [comment]
> First core.windows shall not have a dependency on the editor (that is defined in the ide cluster)

Where can I find this definition?

File: platform/core.windows/nbproject/project.xml [hunk 1]
```diff
@@ -34,6 +34,23 @@
                         <specification-version>3.18</specification-version>
                     </run-dependency>
                 </dependency>
+                <dependency>
+                    <code-name-base>org.netbeans.modules.editor.settings</code-name-base>
```
Comment (lkishalmi) [inline platform/core.windows/nbproject/project.xml:38]: Here is a dependency on editor.settings. That's still part of the ide cluster.

**Comment by matthiasblaesing** [comment]
If I understand if correctly, the API of `core.windows` is provided by `org.openide.windows`. So this is where new API could be established.

I looked into the area and I saw this:

https://github.com/apache/netbeans/blob/12f20e87a763cfa033ab614d7069c006e1c84174/platform/core.windows/src/org/netbeans/core/windows/view/EditorView.java#L201-L213

If you run this:

````
ant -Drun.args.ide=-J-Dnetbeans.winsys.imageSource=org/netbeans/core/windows/resources/topComponentDragMoveFreeArea.gif tryme
```

You get an icon from `core.windows` as background image for the empty editor area. So what could be possible (rough plan to give an idea):

- use the system file system with a folder to lookup the component/component provider to use for background
- modify `EditorView` to lookup the component to use from that folder and replace the code where currently the custom code for `Constants.SWITCH_IMAGE_SOURCE` is used.
- add a special lookup provider that checks for the system property `Constants.SWITCH_IMAGE_SOURCE` and provides a component in the mentioned folder
- add a lookup provider an implementation of your shortcut displayer to the NetBeans branding, where this lookup provider is registered with the path described
- make `EditorView` listen for lookup changes for the chosen path and if the lookup changes let it update the background component

This way you decouple the actual implementation (shortcut display) from the use site (where only a JComponent is expected). The two modules are loosly coupled and other users of the NB Platform can provide their own background providers.

**Comment by neilcsmith-net** [comment]
I really like the idea of doing this, but the implementation here obviously has some issues, and the hardcoded choices are a problem for the platform.  Have had a look at doing something similar to @matthiasblaesing suggestion, but using action lookup rather than allowing for registering any component.

**Comment by Chris2011** [comment]
This PR is obsole due to #7171. Thx @neilcsmith-net

**Comment by tkyjovsk** [comment]
@neilcsmith-net @Chris2011 Hello! Not sure if this is the right place to put feedback. If needed I can create a separate issue for it but for now I'll go ahead and comment here.

I really don't think this is a good idea, in principle. First, if I'm not mistaken there already is a "welcome" dashboard which serves this very purpose.

Second, an empty editor view is _supposed_ to be empty. When all documents are closed there should be no content displayed on the editor pane, imo.

Third, based on the #7171 implementation there seems to be no way to turn this off. At least with the welcome dashboard you are able to disable it once you are familiar with the interface. But now, even if you hide the dashboard you still see this which (at least personally for me) is super-distracting.

**Comment by neilcsmith-net** [comment]
These actions and the dashboard have different purposes.  And there is no reason that because all editors are closed, there should be a massive bit of dead space in the middle of the IDE.  There is no "supposed to be" here!

The implementation was added with the ability to customize, but no UI to do so.

Close NetBeans.  Find your userdir, add a folder at `config/Windows2/`.  Inside that add an empty file with the filename `Background_hidden`. Restart NetBeans.  Problem solved!

A PR to add a UI to customize the images and actions would be considered positively.  It's not on my own immediate radar, but the implementation deliberately allows for it.

**Comment by tkyjovsk** [comment]
@neilcsmith-net Thanks for the off-switch. It works. I appreciate it.

I guess this is a matter of opinion. In my opinion empty space in editor pane signifies that no documents are open. The emptiness by itself has a function and some users (like me) rely on it to convey the information that nothing is open. Empty space doesn't automatically mean "dead space".

If you put some content in a place where a document is supposed to be you indicate that a document is open - without giving the user an option to close it (which for me personally is frustrating). And that something grabs user's attention away from the rest of the UI like the project tree or the file tree. Besides, this only shows a huge icon and a list of actions that are already available on the toolbar - a standard location.

This is a pretty major UI change and in general I don't find it a very good practice to make changes for users without asking if they are interested, and worse, not giving them an (easy) option to say no.

(Please don't take this personally. I know it's not a nice feedback to ask an author how to turn their feature off. I don't mean to be ungrateful, especially on an os project where people put in their own time. I just disagree with the design decision and want to share my pov. I'll try and do the PR if time allows.)

**Comment by Chris2011** [comment]
It is not really an empty editor, it is an empty editor background. So if you would have an empty edtor, you would have a plain file with nothing inside, a tab, line numbers, etc. It is just the place where the editor is supposed to be, when you ot move it around.

Why should not use empty space where we have it? Yes, I can understand and for those who don't want it, we should make an option for this. For this you can create a new ticket or a discussion for that. If you open NetBeans for the very first time as far as I can remember, you see the welcome page and if you close it, you see nothing. So how do you know where to start? What to do? You also don't know what an editor is maybe. It is just a "quick tutorial".

But again let's talk about in a discussion instead of this here.

**Pull request closed without merging**